lib/connection.ts

```diff
@@ -540,24 +628,27 @@ export class Connection extends Entity {
    *
    * @return Promise<AwaitableSender>.
    */
-  async createAwaitableSender(options?: AwaitableSenderOptionsWithSession): Promise<AwaitableSender> {
+  async createAwaitableSender(options?: AwaitableSenderOptionsWithSession & { abortSignal?: AbortSignalLike; }): Promise<AwaitableSender> {
```

Comment:
Why isn't `abortSignal` just in the `AwaitableSenderOptionsWithSession`?

Reply:
`AwaitableSenderOptionsWithSession` extends `AwaitableSenderOptions` which extends `BaseSenderOptions`
`SenderOptionsWithSession` extends `SenderOptions` which extends `BaseSenderOptions`.
`ReceiverOptionsWithSession` extends `ReceiverOptions`

So, the right place to add the abortSignal would have been `BaseSenderOptions` and `ReceiverOptions`

But then, `createRequestResponseLink()` takes both `SenderOptions` and `ReceiverOptions`. Both options supporting `abortSignal` would be not be desirable for this method. So, I could either do the `Pick`/`Omit` magic, or keep `abortSignal` separate as done here. I chose the latter.